File: snakel.py
n  = int(input())
l = [[] for i in range(n)]
xx = n-1
pm = -1
p = 1

for y in range(n):
    for x in range(n):
        l[xx+x*pm].append(p)
        p += 1
    xx += x*pm
    pm = pm * -1

for pl in range(n):
    for ppl in range(n):
        print(l[pl][ppl], end=" ")
    print()

File: snakel.cpp
#include <stdio.h>

int main() {
	int num = 0;
	int p = 1;
	int pm = -1;
	
	scanf("%d", &num);
	int l[num][num];
	int x = num-1;
	
	for(int i = 0; i < num; i++) {
		for(int j = 0; j < num; j++) {
			l[x+j*pm][i] = p;
			p += 1;
		}
		x += (num-1) * pm;
		pm = pm * -1;
	}
	
	for(int pl = 0; pl < num; pl++) {
		for(int ppl = 0; ppl < num; ppl++) {
			printf("%d ",l[pl][ppl]);
		}
		printf("\n");
	}
	
	return 0;
}
